### PR TITLE
USWDS-Compile - Dependencies: POAM March '24

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

Monthly POAM checks and dependency vulnerability resolution.

**Before:** 4 vulnerabilities (1 low, 3 moderate)
**After:** 3 moderate severity vulnerabilities

Updates USWDS package to `3.8.0`

## Related issue

uswds/uswds#5801

Closes https://github.com/uswds/uswds-compile/security/dependabot/10

## Problem statement

Various dependencies were causing **medium** and **low** security vulnerabilities.

## Solution

Bump dependencies with resolving updates.

### Updated dependencies

| Name | Old version | New version |
|:--|:--|:--|
| autoprefixer | `10.4.16` | `10.4.18` |
| postcss | `8.4.32` | `8.4.35` |
| sass-embedded | `1.69.5` | `1.71.1` |
| @uswds/uswds | `^3.7.1` | `3.8.0` [^1] |

[^1]: Note: Pinned the USWDS dependency to match the updating process on USWDS-Site. I figured this grants us more control over breaking changes in the future.

## Testing and review

1. Checkout [test repo](https://github.com/uswds/uswds-sandbox/pull/129)
2. Run `npm install`.
3. Run through gulp commands to confirm they run without error.
4. Run npm start and confirm build completes without error

### Gulp commands

```json
    "uswds:buildDist": "./build.sh",
    "uswds:buildSass": "gulp buildSass",
    "uswds:compileIcons": "gulp compileIcons",
    "uswds:copyAssets": "gulp copyAssets",
    "uswds:copyFonts": "gulp copyFonts",
    "uswds:copyImages": "gulp copyImages",
    "uswds:copyJS": "gulp copyJS",
```

### CODE REVIEW

**Comment by mahoneycm** [comment]
@mejiaj went ahead and updated patch and minor versions as well as updated to `uswds: 3.8.0`.

Tested by installing on sandbox and all compile commands work like a charm with no file changes

File: package.json [hunk 1]
```diff
@@ -33,12 +33,12 @@
...
     "gulp-sourcemaps": "3.0.0",
     "gulp-svgstore": "9.0.0",
-    "postcss": "8.4.32",
+    "postcss": "8.4.35",
     "postcss-csso": "6.0.1",
-    "sass-embedded": "1.69.5"
+    "sass-embedded": "1.71.1"
   },
   "devDependencies": {
-    "@uswds/uswds": "^3.7.1",
```
Comment (mejiaj) [inline package.json:41]: I see your point in having the version number follow the standard convention.

The benefit of `^VERSION_NUMBER` is users will get the latest version without us having to modify this every time.

We'd have to improve our release cadence to make sure users get the latest.

**Review by mejiaj — COMMENTED** [review]
LGTM, thank you. 

A comment on the change in line 41. I'm leaning towards keeping the previous change, but open to making it consistent.

File: package.json (hunk 1 above)
Comment (mahoneycm) [reply package.json:41]: Thanks for the heads up! Resolved in [e2e6588](https://github.com/uswds/uswds-compile/pull/89/commits/e2e6588c17f9fea19970d0262cc9b91902c5b3dd)

**Review by mejiaj — APPROVED** [review]

**Pull request merged**